### PR TITLE
APDS-64 - [BE] add API to logout device

### PR DESCRIPTION
**Pull request by poxip**

### CODE REVIEW

File: rest_framework_jwt/views.py [hunk 1]
```diff
@@ -102,6 +104,25 @@ class RefreshJSONWebToken(JSONWebTokenAPIView):
     serializer_class = RefreshJSONWebTokenSerializer
 
 
+class DeviceLogout(APIView):
+    """
+    Logout user by deleting device.
+    """
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
```
Comment (jacoor) [inline rest_framework_jwt/views.py]: Not get but delete. It is after all delete call.

File: rest_framework_jwt/views.py [hunk 2]
```diff
@@ -102,6 +104,25 @@ class RefreshJSONWebToken(JSONWebTokenAPIView):
...
+        headers = request.META
+        device_name = headers.get('HTTP_X_DEVICE_MODEL', headers.get('HTTP_USER_AGENT'))
+        try:
+            Device.objects.get(
+                name=device_name,
+                user=request.user
+            ).delete()
+        except Device.DoesNotExist:
+            raise NotFound(_('Device does not exist'))
+        return Response({}, status=status.HTTP_200_OK)
```
Comment (jacoor) [inline rest_framework_jwt/views.py]: Delete returns 204 No response.

File: rest_framework_jwt/views.py [hunk 3]
```diff
@@ -102,6 +104,25 @@ class RefreshJSONWebToken(JSONWebTokenAPIView):
...
+    """
+    Logout user by deleting device.
+    """
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+        headers = request.META
+        device_name = headers.get('HTTP_X_DEVICE_MODEL', headers.get('HTTP_USER_AGENT'))
+        try:
+            Device.objects.get(
```
Comment (jacoor) [inline rest_framework_jwt/views.py]: Use some ID that FE will send to find device. Name is not unique.

**Review by jacoor — CHANGES_REQUESTED** [review]

File: rest_framework_jwt/views.py [hunk 4]
```diff
@@ -106,6 +109,23 @@ class RefreshJSONWebToken(JSONWebTokenAPIView):
...
+    """
+    Logout user by deleting Device.
+    """
+    queryset = Device.objects.all()
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        return self.queryset.filter(user=self.request.user)
+
+    def get_object(self):
```
Comment (jacoor) [inline rest_framework_jwt/views.py]: Why two? You have get_queryset and get object. I'd use only get object, there is no point in keeping two functions.

**Review by jacoor — CHANGES_REQUESTED** [review]

File: rest_framework_jwt/views.py (hunk 4 above)
Comment (poxip) [reply rest_framework_jwt/views.py]: But the get_object() uses the get_queryset(), but yeah, it can be done in the get_object()

Comment (poxip) [reply rest_framework_jwt/views.py]: Covered in the latest CR commit.

**Pull request merged**